File: REST/core/e3.pl
#!/bin/env perl

# Ping the server 25 times to see how rate limiting works

use strict;
use warnings;

use HTTP::Tiny;
use JSON;
use Data::Dumper;
$Data::Dumper::Terse = 1;
$Data::Dumper::Indent = 1;

my $server = "http://rest.ensembl.org";
my $ping_endpoint = "/info/ping";

my $http = HTTP::Tiny->new();

# Fetch an endpoint from the serverm allow overriding of the default content type
# Don't do error checking this time and return the response as-is
sub fetch_endpoint {
    my ($request, $content_type) = @_;
    $content_type ||= 'application/json';

    my $response = $http->get($request,
                              { headers => { 'Content-type' => $content_type } } );

    return $response;

}

print "Accessing ping endpoint\n";

for(my $i = 1; $i <= 25; $i++) {
    my $r = fetch_endpoint($server.$ping_endpoint);

    print "Count $i, status: " . $r->{status} . "\n";
    print "\tRemaining limit: " . $r->{headers}->{'x-ratelimit-remaining'} . "\n";
}

File: REST/core/e3.py
import requests, sys
import json
import pprint

'''
Ping the server 25 times to see how rate limiting works
'''

server = "http://rest.ensembl.org"
ping_endpoint = "/info/ping"

def fetch_endpoint(server, request, content_type='application/json'):
    """
    Fetch an endpoint from the server, allow overriding of default content-type
    Don't do error checking this time and return the response as-is
    """
    r = requests.get(server+request, headers={ "Content-Type" : content_type})

    return r

if __name__ == "__main__":

    print "Accessing ping endpoint"

    for i in range(1, 26):
        r = fetch_endpoint(server, ping_endpoint)
        print "Count {}, status: {}".format(i, r.status_code)
        print "\tRemaining limit: {}".format(r.headers['X-RateLimit-Remaining'])
